src/server.ts

```diff
@@ -191,18 +191,29 @@ let rc_file = path.join(config.home_dir, '.analysis_rc');
 if (fs.existsSync(rc_file)) {
     let rc = JSON.parse(fs.readFileSync(rc_file, 'utf8'));
     if ('server' in rc) {
-        config.server_url = `${rc.server}/api/v1`;
+        config.server_url = `${rc.server}/api/v2`;
     }
 }
 
 let DiagnosticsEngines = [SecurityEngine];
 
 const getCAmsg = (deps, diagnostics): string => {
+    let msg : string;
     if(diagnostics.length > 0) {

```

Comment:
`diagnostics` array is a type of `Diagnostic` and it has pvtVul & pubVul fields. You can use those to count.

Reply:
Diagnostic is not having any fields for vulnerability count, but it is a part of Diagnostic.message. 

here : https://github.com/dvandra/fabric8-analytics-lsp-server/blob/LSP-Changes-for-Component-Analysis-API-V2/src/consumers.ts#L186

Reply:
Actually you have to refer `DiagnosticsPipeline` object, it has `items` field which is a type of `SecurityEngine`. `SecurityEngine` engine has necessary fields to count vulnerabilities.

Reply:
Something like codeActionsMap to collect the codeActions[].
Correct?

Reply:
@arajkumar PTAL : https://github.com/fabric8-analytics/fabric8-analytics-lsp-server/pull/116/commits/6abe935b8310d83a8b579797ddde74ce039672eb